lib/models/event-timeline.js

```diff
@@ -25,16 +25,17 @@ var MatrixEvent = require("./event").MatrixEvent;
  * <p>Once a timeline joins up with its neighbour, they are linked together into a
  * doubly-linked list.
  *
- * @param {string} roomId    the ID of the room where this timeline came from
+ * @param {EventTimelineSet} eventTimelineSet the set of timelines this is part of
  * @constructor
  */
-function EventTimeline(roomId) {
-    this._roomId = roomId;
+function EventTimeline(eventTimelineSet) {
+    this._eventTimelineSet = eventTimelineSet;
+    this._roomId = eventTimelineSet.roomId;
     this._events = [];
     this._baseIndex = 0;
-    this._startState = new RoomState(roomId);
+    this._startState = new RoomState(this._roomId);
```

Comment:
presumably `this._roomId` is going to be null for the notifications timeline. `RoomState`'s jsdoc explicitly says that the roomId is required, and a brief skim through the implementation doesn't convince me that it is lying. Have you checked what will happen here?